crash after calibration: accuracy visualizer

Experienced a crash the other day after calibration (the data for calibration was not good enough for sure), here is the traceback that I saved:

``` 
world - [ERROR] launchables.world: Process Capture crashed with trace:
Traceback (most recent call last):
  File "/scratch/neuromod/src/pupil/pupil_src/launchables/world.py", line 742, in world
    p.on_notify(n)
  File "/scratch/neuromod/src/pupil/pupil_src/shared_modules/accuracy_visualizer.py", line 302, in on_notify
    if self.__handle_calibration_result_notification(notification):
  File "/scratch/neuromod/src/pupil/pupil_src/shared_modules/accuracy_visualizer.py", line 336, in __handle_calibration_result_notification
    self.recalculate()
  File "/scratch/neuromod/src/pupil/pupil_src/shared_modules/accuracy_visualizer.py", line 403, in recalculate
    hull = ConvexHull(ref_locations)  # requires at least 3 points
  File "qhull.pyx", line 2431, in scipy.spatial.qhull.ConvexHull.__init__
  File "qhull.pyx", line 356, in scipy.spatial.qhull._Qhull.__init__
scipy.spatial.qhull.QhullError: QH6013 qhull input error: input is less than 2-dimensional since all points have the same x coordinate  0.5

While executing:  | qhull i Qt
Options selected for Qhull 2019.1.r 2019/06/21:
  run-id 150558979  incidence  Qtriangulate  _pre-merge  _zero-centrum
  _max-width 0.9  Error-roundoff 8.1e-16  _one-merge 4.1e-15
  _near-inside 2e-14  Visible-distance 1.6e-15  U-max-coplanar 1.6e-15
  Width-outside 3.3e-15  _wide-facet 9.8e-15  _maxoutside 4.9e-15

``` 

I think that this exception should be caught to avoid crashing the whole software, WDYT? 

Thanks

--- a/pupil_src/shared_modules/accuracy_visualizer.py
+++ b/pupil_src/shared_modules/accuracy_visualizer.py
@@ -10,13 +10,13 @@ See COPYING and COPYING.LESSER for license details.
 """
 
 import logging
-from collections import namedtuple
 import typing as T
+from collections import namedtuple
 
-import OpenGL.GL as gl
 import numpy as np
+import OpenGL.GL as gl
 from pyglui import ui
-from pyglui.cygl.utils import draw_points_norm, draw_polyline_norm, RGBA
+from pyglui.cygl.utils import RGBA, draw_points_norm, draw_polyline_norm
 from scipy.spatial import ConvexHull
 
 from calibration_choreography import (
@@ -24,15 +24,13 @@ from calibration_choreography import (
     ChoreographyMode,
     ChoreographyNotification,
 )
-from plugin import Plugin
-
 from gaze_mapping import gazer_classes_by_class_name, registered_gazer_classes
 from gaze_mapping.notifications import (
-    CalibrationSetupNotification,
     CalibrationResultNotification,
+    CalibrationSetupNotification,
 )
 from gaze_mapping.utils import closest_matches_monocular
-
+from plugin import Plugin
 
 logger = logging.getLogger(__name__)
 

--- a/pupil_src/shared_modules/accuracy_visualizer.py
+++ b/pupil_src/shared_modules/accuracy_visualizer.py
@@ -11,7 +11,6 @@ See COPYING and COPYING.LESSER for license details.
 
 import logging
 import typing as T
-from collections import namedtuple
 
 import numpy as np
 import OpenGL.GL as gl

--- a/pupil_src/shared_modules/accuracy_visualizer.py
+++ b/pupil_src/shared_modules/accuracy_visualizer.py
@@ -10,13 +10,14 @@ See COPYING and COPYING.LESSER for license details.
 """
 
 import logging
+import traceback
 import typing as T
 
 import numpy as np
 import OpenGL.GL as gl
+import scipy.spatial
 from pyglui import ui
 from pyglui.cygl.utils import RGBA, draw_points_norm, draw_polyline_norm
-from scipy.spatial import ConvexHull
 
 from calibration_choreography import (
     ChoreographyAction,
@@ -397,8 +398,13 @@ class Accuracy_Visualizer(Plugin):
         self.error_lines = results.error_lines
         ref_locations = results.correlation.norm_space[1::2, :]
         if len(ref_locations) >= 3:
-            hull = ConvexHull(ref_locations)  # requires at least 3 points
-            self.calibration_area = hull.points[hull.vertices, :]
+            try:
+                # requires at least 3 points
+                hull = scipy.spatial.ConvexHull(ref_locations)
+                self.calibration_area = hull.points[hull.vertices, :]
+            except scipy.spatial.qhull.QhullError:
+                logger.warning("Calibration area could not be calculated")
+                logger.debug(traceback.format_exc())
 
     @staticmethod
     def calc_acc_prec_errlines(

